src/pages/product-list-page/product-list-filter/product-list-filter.js

```diff
@@ -0,0 +1,328 @@
+      Products: { products },
+      Filter: {
+        currentPage,
+        productsPerPage,
+        sortByPrice,
+        sortByRate,
+        sortByPopularity
+      },
+      Language: { language }
+    }) => ({
+      products,
+      currentPage,
+      productsPerPage,
+      sortByPrice,
+      sortByRate,
+      sortByPopularity,
+      language
+    })
+  );
+
+  useEffect(() => {
+    setPrice([
+      Math.min(...products.map((product) => product.basePrice)),
+      Math.max(...products.map((product) => product.basePrice))
+    ]);
+    setCategoryCheck({ [selectedCategory]: true });
+  }, [products, selectedCategory]);
+
+  const colors = [
+    ...new Set(
+      products.map((product) => JSON.stringify(product.colors[0].simpleName))
+    )
+  ].map(JSON.parse);
+  const categories = [
+    ...new Set(products.map((product) => JSON.stringify(product.category)))
+  ].map(JSON.parse);
+
+  const patterns = [
+    ...new Set(
+      products.map((product) =>
+        product.pattern.length
+          ? JSON.stringify(product.pattern)
+          : JSON.stringify([
+            {
+              lang: 'uk',
+              value: 'Немає'
+            },
+            {
+              lang: 'eng',
+              value: 'None '
+            }
+          ])
+      )
+    )
+  ].map(JSON.parse);
+  const [search, setSearch] = useState('');
+  const [price, setPrice] = useState([null, null]);
+  const [categoryCheck, setCategoryCheck] = useState({});
+  const [colorsCheck, setColorsCheck] = useState({});
+  const [patternsCheck, setPatternsCheck] = useState({});
+
+  const handleSearch = (event) => {
+    setSearch(event.target.value);
+  };
+
+  const handlePriceChange = (event, newValue) => {
+    setPrice(newValue);
+  };
+
+  const handleCategoryChange = (event) => {
+    setCategoryCheck({
+      ...categoryCheck,
+      [event.target.name]: event.target.checked
+    });
+  };
+
+  const handleColorChange = (event) => {
+    setColorsCheck({
+      ...colorsCheck,
+      [event.target.name]: event.target.checked
+    });
+  };
+
+  const handlePatternChange = (event) => {
+    setPatternsCheck({
+      ...patternsCheck,
+      [event.target.name]: event.target.checked
+    });
+  };
+
+  const handleFilter = () => {
+    dispatch(
+      setColorsFilter(
+        colors
+          .filter((color) => colorsCheck[color[1].value])
+          .map((color) => color[1].value)
+      )
+    );
+    dispatch(
+      setPatternsFilter(
+        patterns
+          .filter((pattern) => patternsCheck[pattern[1].value])
+          .map((pattern) => pattern[1].value)
+      )
+    );
+    dispatch(
+      setCategoryFilter(
+        categories
+          .filter((category) => categoryCheck[category.name[1].value])
+          .map((category) => category._id)
+      )
+    );
+    dispatch(setSearchFilter(search));
+    dispatch(setPriceFilter(price));
+    dispatch(
+      getFiltredProducts({
+        search,
+        price,
+        colors: colors
```

Comment:
It should be done inside a saga where you take filter object from the `redux` state and make request

Reply:
fixed